src/Tribe/Events/Attendees_List.php

```diff
@@ -68,20 +90,31 @@ public function is_event_hiding_attendee_list( $post ) {
 	 * @since TBD
 	 *
 	 * @param bool $should_hide Whether the optout form should be hidden or not.
+	 * @param int  $post_id     The post ID the ticket belongs to.
 	 *
 	 * @return bool Whether we should hide the optout option.
 	 *
 	 * @see \Tribe\Tickets\Events\Events_Service_Provider::hooks
 	 */
-	public function should_hide_optout( $should_hide ) {
-		try {
-			global $post;
+	public function should_hide_optout( $should_hide, $post_id = 0 ) {
+		global $post;
+
+		return true;
+
+		if ( empty( $post_id ) ) {
+			$post_id = $post;
+		}
+
+		$is_hidden_on = static::is_hidden_on( $post_id, false );
 
-			return $this->is_event_hiding_attendee_list( $post );
-		} catch ( Exception $e ) {
-			// Eg: global $post not a WP_Post object.
-			return $should_hide;
+		if ( null !== $is_hidden_on ) {
+			return $is_hidden_on;
 		}
+
+		$post = get_post( $post_id );
+
+		// The setting isn't set yet, but let's double check until the setting gets migrated.
+		return ! $this->is_showing_attendee_list_with_blocks( $post );
```

Comment:
:warning: Code after the RETURN statement on line 102 cannot be executed

<sub>hash: e05fc39f8a24d13646c2cf691195e315</sub>